### PR TITLE
[Aftershock] Arcology First Floor

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Mods "Aftershock Arcology First Floor Draft"

#### Purpose of change
Add's a new city special for Aftershock.  Currently intended to fill several roles and partially replace existing lab types.  Will have possibility for finale rooms on upper and lower floors.  

#### Describe the solution
First floor will be a mall like space with ruined stores.   Floors going up will be mostly living space/administration/white collar.  Floors going down will be mostly manufacturing/security related floors.  Once tests pass this will be ready to go in.  This is a shell only at this point and needs hundreds of additional lines of variants for the first floor alone.

#### Describe alternatives you've considered
Making it even bigger.  Realistically there are no other alternatives to creating massive amounts of content to enable Aftershock to become a total conversion mod.  

#### Testing
Everything loads without errors.   Significant filler was added to get to that point but this PR is already three times the size of a reasonable PR even for JSON.  Need to add some micro mills to the exterior of the arcology. Both types once the materials PR merges.  Also wilderness ready vehicles. 
gates console

Describe what steps you took to test that this PR resolved the bug or added the feature, and what tests you performed to make sure it didn't cause any regressions.  Also include testing suggestions for reviewers and maintainers. -->

### CODE REVIEW

File: data/mods/Aftershock/itemgroups/arcology_groups.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[
+  {
+    "id": "vitamin_shop",
+    "type": "item_group",
+    "items": [
+      [ "cream_prot_cold", 25 ],
+      [ "cream_greater_prot_cold", 5 ],
+      [ "homeopathic_pills", 80 ]
+    ]
+  },
```
Comment (Maleclypse) [inline data/mods/Aftershock/itemgroups/arcology_groups.json:10]:
```suggestion
  }
```

File: data/mods/Aftershock/maps/mapgen/arcology_ground.json [hunk 2]
```diff
@@ -0,0 +1,835 @@
...
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": [ "arcology_exterior_e_side_entrance_robots", "arcology_exterior_e_side_entrance_uplifts" ],
+    "weight": 100,
+    "object": {
+      "rows": [
+        "________________óó,,,,,,",
+        "_________1______||,,,,,,",
```
Comment (Maleclypse) [inline data/mods/Aftershock/maps/mapgen/arcology_ground.json:351]: Error here.
12:58:10.277 ERROR : src/main_menu.cpp:863 [bool main_menu::new_character_tab()] Error: Json error: data/mods/Aftershock/maps/mapgen/arcology_ground.json:351:19: format: rows: row 2 column 10: '1' is not in 'terrain', and no 'fill_ter' is set!

      "rows": [
        "________________óó,,,,,,",
        "_________1
                  ^
                   ______||,,,,,,",
        "________________||,,,,,,",
        "________________||,,,,,,",

12:58:42.065 : Log shutdown.

**Comment by Maleclypse** [comment]
@curstwist got the console commands working!!!

File: .gitignore [hunk 3]
```diff
@@ -204,3 +204,4 @@ Xcode/
 /tests/pch/tests-pch.hpp.pch
 /tests/catch/catch.hpp.gch
 /tests/catch/catch.hpp.pch
+vcpkg/
```
Comment (Maleclypse) [inline .gitignore:207]:
```suggestion

```

File: data/mods/Aftershock/maps/overmap_terrain.json [hunk 4]
```diff
@@ -234,5 +234,69 @@
...
+      "arcology_exterior_ne_corner_robots",
+      "arcology_exterior_nw_corner_robots",
+      "arcology_exterior_se_corner_robots",
+      "arcology_exterior_sw_corner_robots",
+      "arcology_exterior_n_entrance_robots",
+      "arcology_exterior_s_entrance_robots",
+      "arcology_exterior_e_entrance_robots",
+      "arcology_exterior_w_entrance_robots"
+    ],
+    "name": "Arcology Ground Level (no life signs)",
```
Comment (John-Candlebury) [inline data/mods/Aftershock/maps/overmap_terrain.json:263]: Sort of feel these break the 4th wall a bit. Maybe just name it
```suggestion
    "name": "Desolate Arcology - Ground Level",
```

File: data/mods/Aftershock/maps/overmap_terrain.json [hunk 5]
```diff
@@ -234,5 +234,69 @@
...
+      "arcology_exterior_ne_corner_uplifts",
+      "arcology_exterior_nw_corner_uplifts",
+      "arcology_exterior_se_corner_uplifts",
+      "arcology_exterior_sw_corner_uplifts",
+      "arcology_exterior_n_entrance_uplifts",
+      "arcology_exterior_s_entrance_uplifts",
+      "arcology_exterior_e_entrance_uplifts",
+      "arcology_exterior_w_entrance_uplifts"
+    ],
+    "name": "Arcology Ground Level (life signs)",
```
Comment (John-Candlebury) [inline data/mods/Aftershock/maps/overmap_terrain.json:295]: Same as the above
```suggestion
    "name": "Arcology - Ground Level",
```

**Comment by anothersimulacrum** [comment]
There are merge conflicts.

**Pull request merged**